gremlin-scala/src/test/scala/gremlin/scala/TraversalSpec.scala

```diff
@@ -394,6 +395,34 @@ class TraversalSpec extends WordSpec with Matchers {
     }
   }
 
+  "choose" should {
+    "work for if/then/else semantics" in new Fixture {
+
+      val traversal: GremlinScala[String, _] = graph
+        .V.hasLabel("person")
+        .choose(
+          _.values("age").is(P.lte(30)),
```

Comment:
you could use the (typed) property Key as defined at the bottom and used in other tests here, i.e. `_.value2(Age)`